File: algorithms/implementation/append-and-delete/append-and-delete-python/tests/test_appendAndDelete.py
from appendAndDelete import appendAndDelete

class TesteAppendAndDelete:
    
    def testeCaso1(self):
        
        s = 'hackerhappy'
        t = 'hackerrank'
        k = 9

        esperado = 'Yes'
        resultado = appendAndDelete(s, t, k)
        
        assert resultado == esperado
        
    def testeCaso2(self):
        
        s = 'aba'
        t = 'aba'
        k = 7

        esperado = 'Yes'
        resultado = appendAndDelete(s, t, k)
        
        assert resultado == esperado
        
    def testeCaso3(self):
        
        s = 'aaaaaaaaaa'
        t = 'aaaaa'
        k = 7

        esperado = 'Yes'
        resultado = appendAndDelete(s, t, k)
        
        assert resultado == esperado
        
        
    def testeCaso4(self):
        
        s = 'ashley'
        t = 'ash'
        k = 2

        esperado = 'No'
        resultado = appendAndDelete(s, t, k)
        
        assert resultado == esperado

File: algorithms/implementation/append-and-delete/append-and-delete-javascript/js/test_appendAndDelete.js

import {appendAndDelete} from '../js/appendAndDelete.js';
import assert from 'assert';

describe('appendAndDelete', function() {

    it('É possível converter s para t com número mínimo de operações necessárias?: Yes', function() {

        let s = 'hackerhappy';
        let t = 'hackerrank';
        let k = 9;

        let esperado = 'Yes';
        let resultado = appendAndDelete(s, t, k);

        assert.strictEqual(resultado, esperado)


    });
});

describe('appendAndDelete', function() {

    it('É possível converter s para t com número mínimo de operações necessárias?: Yes', function() {

        let s = 'aba';
        let t = 'aba';
        let k = 7;

        let esperado = 'Yes';
        let resultado = appendAndDelete(s, t, k);

        assert.strictEqual(resultado, esperado)


    });
});

describe('appendAndDelete', function() {

    it('É possível converter s para t com número mínimo de operações necessárias?: Yes', function() {

        let s = 'aaaaaaaaaa';
        let t = 'aaaaa';
        let k = 7;

        let esperado = 'Yes';
        let resultado = appendAndDelete(s, t, k);

        assert.strictEqual(resultado, esperado)


    });
});

describe('appendAndDelete', function() {

    it('É possível converter s para t com número mínimo de operações necessárias?: No', function() {

        let s = 'ashley';
        let t = 'ash';
        let k = 2;

        let esperado = 'No';
        let resultado = appendAndDelete(s, t, k);

        assert.strictEqual(resultado, esperado)


    });
});